app/controllers/developer_controller.rb

```diff
@@ -13,7 +13,18 @@ def update_sites
 
   def production_log
     @log = if params[:grep].present?
-             `tail -n 10000 log/production.log | grep -E '#{params[:grep]}' -C #{params[:context]} --color=never`
+             begin
+               grep_str = params[:grep].instance_of?(String) ? params[:grep] : params[:grep].to_s
+               context_int = params[:context].instance_of?(Integer) ? params[:context] : params[:context].to_i
+               # Check if coercion is successful, in case someone overloaded .to_*() methods
+               if !grep_str.instance_of?(String) || !context_int.instance_of?(Integer)
+                 raise ArgumentError, 'Coercion failure'
```

Comment:
Raising in a rescue is a very weird thing to do and is almost never a good thing, especially since rescuing only re-raises a different error.

Reply:
The intention of `rescue` is mainly to catch `Missing method` errors raised by `.to_*()`, as we probably can't assume `params` is a dictionary of strings (may have nested dictionaries, for example); the `raise ArgumentError` is a reuse of the `rescue` statement in an effort to make the code less redundant.

Reply:
If you want to catch Method Missing errors, use `respond_to?` instead. It's a much more standard way of checking if an object responds to a method.